azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -328,34 +328,20 @@ def start_extension_command(self, extension_name, command, timeout, shell, cwd,
             :param stderr: File object to redirect stderr to
             :param error_code: Extension error code to raise in case of error
             """
-            if not self.enabled():
-                # subprocess-popen-preexec-fn<W1509> Disabled: code is not multi-threaded
-                process = subprocess.Popen(command,  # pylint: disable=W1509
-                                           shell=shell,
-                                           cwd=cwd,
-                                           env=env,
-                                           stdout=stdout,
-                                           stderr=stderr,
-                                           preexec_fn=os.setsid)
-
-                process_output = handle_process_completion(process=process,
-                                                           command=command,
-                                                           timeout=timeout,
-                                                           stdout=stdout,
-                                                           stderr=stderr,
-                                                           error_code=error_code)
-            else:
-                process_output = self._cgroups_api.start_extension_command(extension_name,
-                                                                          command, 
-                                                                          timeout, 
-                                                                          shell=shell, 
-                                                                          cwd=cwd, 
-                                                                          env=env, 
-                                                                          stdout=stdout, 
-                                                                          stderr=stderr, 
-                                                                          error_code=error_code) 
-
-            return process_output
+            if self.enabled():
+                try:
+                    return self._cgroups_api.start_extension_command(extension_name, command, timeout, shell=shell, cwd=cwd, env=env, stdout=stdout, stderr=stderr, error_code=error_code)
```

Comment:
Previously CGroupsApi would do the retry. I moved the retry logic here and nor the CGroupsConfigurator does the retry after disabling cgroups.